### PR TITLE
Created transactions page

### PR DESCRIPTION
**Pull request by aryanxved**
- The transactions page has been implemented 
- New reusableComponents folder has been created for components that we will commonly use
- Transaction page supports dark mode
- Updated header styling on homepage

### CODE REVIEW

File: app/(auth)/(tabs)/transactions.tsx [hunk 1]
```diff
@@ -1,102 +1,308 @@
...
+
+  // Render the home screen 
   return (
-    <ParallaxScrollView
-      headerBackgroundColor={{ light: '#D0D0D0', dark: '#353636' }}
-      headerImage={<Ionicons size={310} name="code-slash" style={styles.headerImage} />}>
-      <ThemedView style={styles.titleContainer}>
-        <ThemedText type="title">Transaction</ThemedText>
+    <SafeAreaView style={[styles.main, {backgroundColor: theme.colors.background}]}>
+      <ScrollView style={[styles.mainContainer, {backgroundColor: theme.colors.background}]}>
```
Comment (SaifA21) [inline app/(auth)/(tabs)/transactions.tsx]: We should have a 'sticky' header to indicate what page you are on i.e. having the scrollview only in the view where we render the transactions, right now it looks odd to have the header scroll out of view 

https://github.com/user-attachments/assets/7b271f2e-996a-45ab-88ff-394812e59c7e

File: app/(auth)/(tabs)/transactions.tsx [hunk 2]
```diff
@@ -1,102 +1,308 @@
...
+        </ThemedView>
+        <ThemedText style={styles.subHeadingText}>TRANSACTION PREVIEW</ThemedText>
+        <ThemedView>
+          {renderTransactionPreview()}
+          </ThemedView>
+
+        <ThemedText style={styles.subHeadingText}>TRANSACTION LIST</ThemedText>
+
+
+      <ThemedView>
```
Comment (SaifA21) [inline app/(auth)/(tabs)/transactions.tsx:141]: this can be our scroll view

**Review by SaifA21 — CHANGES_REQUESTED** [review]

File: app/(auth)/(tabs)/transactions.tsx (hunk 1 above)
Comment (aryanxved) [reply app/(auth)/(tabs)/transactions.tsx]: Weirdly enough, the sticky header is what I'm working on right now. Right before your comment.

File: app/(auth)/(tabs)/transactions.tsx (hunk 2 above)
Comment (aryanxved) [reply app/(auth)/(tabs)/transactions.tsx:141]: Trying a few styles out. I think that the transaction preview should be included in the scroll view and we have a common page header. Committing an update real quick.

**Review by SaifA21 — APPROVED** [review]

**Pull request merged**